lib/util/cache.js

```diff
@@ -0,0 +1,178 @@
+          // If we're given multiple headers, the comma messes us up since
+          //  we split the full header by commas. So, let's loop through the
+          //  remaining parts in front of us until we find one that ends in a
+          //  quote. We can then just splice all of the parts in between the
+          //  starting quote and the ending quote out of the directives array
+          //  and continue parsing like normal.
+          // https://www.rfc-editor.org/rfc/rfc9111.html#name-no-cache-2
+          if (value[0] === '"') {
+            // Something like `no-cache="some-header"` OR `no-cache="some-header, another-header"`.
+
+            // Add the first header on and cut off the leading quote
+            const headers = [value.substring(1)]
+
+            let foundEndingQuote = value[value.length - 1] === '"'
+            if (!foundEndingQuote) {
+              // Something like `no-cache="some-header, another-header"`
+              //  This can still be something invalid, e.g. `no-cache="some-header, ...`
+              for (let j = i + 1; j < directives.length; j++) {
+                const nextPart = directives[j]
+                const nextPartLength = nextPart.length
+
+                headers.push(nextPart.trim())
+
+                if (nextPartLength !== 0 && nextPart[nextPartLength - 1] === '"') {
+                  foundEndingQuote = true
+                  break
+                }
+              }
+            }
+
+            if (foundEndingQuote) {
+              let lastHeader = headers[headers.length - 1]
+              if (lastHeader[lastHeader.length - 1] === '"') {
+                lastHeader = lastHeader.substring(0, lastHeader.length - 1)
+                headers[headers.length - 1] = lastHeader
+              }
+
+              output[key] = headers
+            }
+          } else {
+            // Something like `no-cache=some-header`
+            output[key] = [value]
+          }
+
+          break
+        }
+      }
+      // eslint-disable-next-line no-fallthrough
+      case 'public':
+      case 'no-store':
+      case 'must-revalidate':
+      case 'proxy-revalidate':
+      case 'immutable':
+      case 'no-transform':
+      case 'must-understand':
+      case 'only-if-cached':
+        if (value) {
+          continue
+        }
+
+        output[key] = true
+        break
+      default:
+        // Ignore unknown directives as per https://www.rfc-editor.org/rfc/rfc9111.html#section-5.2.3-1
+        continue
+    }
+  }
+
+  return output
+}
+
+/**
+ * @param {string} varyHeader Vary header from the server
+ * @param {Record<string, string>} headers Request headers
+ * @returns {Record<string, string>}
+ */
+function parseVaryHeader (varyHeader, headers) {
+  const output = /** @type {Record<string, string>} */ ({})
+
+  const varyingHeaders = varyHeader.toLowerCase().split(',')
+  for (const header of varyingHeaders) {
+    const trimmedHeader = header.trim()
```

Comment:
what if varyHeader is `*` which means all headers?

Reply:
There's an early return point elsewhere that should prevent a response with `vary: *` from getting to this point, definitely a good idea to still add a check though